app/src/main/kotlin/com/hippo/ehviewer/EhDB.kt

```diff
@@ -142,6 +146,14 @@ object EhDB {
         dao.fill(raw.position)
     }
 
+    suspend fun putDownloadArtist(gid: Long, artists: List<DownloadArtist>) {
+        val dao = db.downloadArtistDao()
+        dao.deleteByGid(gid)
```

Comment:
Move these inside if.

Reply:
Should we ensure that it is updated, even if the artists is empty?

Reply:
Atrists should never become empty from non-empty.
When it happens, it usually means something went wrong during processing and we don't want valid data to be overwritten.